packages/mobile/routes/dashboard/wallet/drawers/account-switcher/AccountSwitcherDrawer.svelte

```diff
@@ -0,0 +1,57 @@
+<script lang="typescript">
+    import { localize } from '@core/i18n'
+    import { tryCreateAdditionalAccount } from '@core/account'
+    import { isStrongholdUnlocked } from '@core/profile-manager'
+    import { Drawer, StrongholdUnlock } from '../../../../../components'
+    import { AccountSwitcherRoute, accountSwitcherRoute, accountSwitcherRouter } from '../../../../../lib/core/router'
+    import { onDestroy } from 'svelte'
+    import { AccountSwitcher, CreateAccount } from './views'
+
+    export let onClose: () => unknown = () => {}
+
+    let accountAlias = ''
+    let color = ''
+
+    const titles = {
+        [AccountSwitcherRoute.Switcher]: localize('general.accounts'),
+        [AccountSwitcherRoute.CreateAccount]: localize('general.createAccount'),
+    }
+
+    async function onCreate(newAccountAlias: string, newColor: string): Promise<void> {
+        accountAlias = newAccountAlias
+        color = newColor
+
+        const isUnlocked = await isStrongholdUnlocked()
+        if (isUnlocked) {
+            await tryCreateAdditionalAccount(newAccountAlias, newColor)
+            onClose()
+        } else {
+            $accountSwitcherRouter.next()
+        }
+    }
+
+    async function onUnlock() {
+        await tryCreateAdditionalAccount(accountAlias, color)
+        onClose()
+    }
+
+    onDestroy(() => {
+        $accountSwitcherRouter.reset()
+    })
+</script>
+
+<Drawer {onClose} title={titles[$accountSwitcherRoute]}>
```

Comment:
I guess we need a flag for the drawer to prevent it from being closed when the content is busy